src/services/identity/CollaboratorsService.ts

```diff
@@ -235,6 +238,39 @@ class CollaboratorsService {
 
     return (site?.site_members?.[0]?.SiteMember?.role as string | null) ?? null
   }
+
+  getStatistics = async (siteName: string) => {
+    const inactiveLimit = new Date()
+    inactiveLimit.setDate(
+      inactiveLimit.getDate() - INACTIVE_USER_THRESHOLD_DAYS
+    )
+    const site = await this.siteRepository.findOne({
+      where: { name: siteName },
+      include: [
+        {
+          model: User,
+          as: "site_members",
+        },
+      ],
+    })
+
+    const collaborators = site?.site_members ?? []
+    const totalCount = collaborators.length
+
+    if (totalCount === 0) {
+      // Every site must have at least one collaborator
+      return new NotFoundError(`Site does not exist`)
+    }
```

Comment:
wouldn't the correct condition be `!site`?

Reply:
Hmm this condition is wider because earlier I checked if `site` and `site.site_members` exists. Beyond that, in the unlikely event that the site exists but there are no collaborators, this condition still catches it correctly.